proxy/src/cplane_api.rs

```diff
@@ -1,106 +1,85 @@
-use anyhow::{anyhow, bail, Context};
+use crate::auth::ClientCredentials;
+use crate::compute::DatabaseInfo;
+use crate::waiters::{Waiter, Waiters};
+use anyhow::{anyhow, bail};
+use lazy_static::lazy_static;
 use serde::{Deserialize, Serialize};
-use std::net::{SocketAddr, ToSocketAddrs};
 
-use crate::state::ProxyWaiters;
-
-#[derive(Serialize, Deserialize, Debug, Default)]
-pub struct DatabaseInfo {
-    pub host: String,
-    pub port: u16,
-    pub dbname: String,
-    pub user: String,
-    pub password: Option<String>,
+lazy_static! {
+    static ref CPLANE_WAITERS: Waiters<Result<DatabaseInfo, String>> = Default::default();
```

Comment:
I wonder is there any noticeable difference with using wakers directly instead of oneshots? With oneshots it is just less amount of code, but wakers could be cheaper. Thoughts?

Reply:
I don't think this change is necessary here. I'll reconsider this if it ever becomes a bottleneck.